src/services/expenses.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const { generateId } = require('../helpers/generateId');
+
+let expenses = [];
+
+function setInitialExpenses() {
+  expenses = [];
+};
+
+function getAllExpenses(searchParams) {
+  const { userId, category, from, to } = searchParams;
+
+  return expenses.filter(expense => {
+    const isUserIdMatch = userId
+      ? expense.userId === Number(userId)
+      : true;
+
+    const areCategoriesMatch = category
+      ? expense.category === category
+      : true;
+
+    const isFromMatch = from
+      ? expense.spentAt >= from
+      : true;
+
+    const isToMatch = to
+      ? expense.spentAt <= to
+      : true;
+
+    return isUserIdMatch && areCategoriesMatch && isFromMatch && isToMatch;
+  });
+}
+
+function getExpenseById(expenseId) {
```

Comment:
This is service only for expences, so no need to use full name, you can just write `getById`